lib/afinter.c

```diff
@@ -635,3 +641,107 @@ afinter_get_metrics(void)
 {
   return metrics;
 }
+
+/****************************************************************************
+ * Live collection of internal log messages.
+ ****************************************************************************/
+
+AFInterLive
+afinter_start_live_collection(void)
+{
+  AFInterSource *self;
+
+  if (is_live_collection)
+    return AFINTER_LIVE_COLLECTION_RUNNING;
+
+  if (current_internal_source != NULL)
+    return AFINTER_INTERNAL_SRC_PRESENT;
+
+  self = g_new0(AFInterSource, 1);
+  AFInterSourceOptions *options = g_new0(AFInterSourceOptions, 1);
+
+  afinter_source_options_defaults(options);
+  self->options = options;
+  stats_counter_set(metrics.queue_capacity, self->options->queue_capacity);
+
+  g_mutex_lock(&internal_msg_lock);
+  current_internal_source = self;
+  g_mutex_unlock(&internal_msg_lock);
+
+  is_live_collection = TRUE;
+
+  return AFINTER_LIVE_COLLECTION_STARTED;
+}
+
+void
+afinter_reset_live_collection(void)
+{
+  afinter_stop_live_collection();
+
+  if (is_live_collection)
+    {
+      if (internal_msg_queue)
+        _release_internal_msg_queue();
+    }
+}
+
```

Comment:
I guees `afinter_stop_live_collection()` must also be called in somewhere in the cleanup chain e.g. in `afinter_sd_deinit()`

Reply:
@HofiOne : okay, done 👍